db/background_mgr_attachment_compaction.go

```diff
@@ -142,9 +191,30 @@ func (a *AttachmentCompactionManager) Run(ctx context.Context, options map[strin
 		fallthrough
 	case "cleanup":
 		a.SetPhase("cleanup")
-		persistClusterStatus()
-		err := attachmentCompactCleanupPhase(ctx, dataStore, collectionID, database, a.CompactID, a.VBUUIDs, terminator)
+		worker := func() (shouldRetry bool, err error, value interface{}) {
+			persistClusterStatus()
+			err = attachmentCompactCleanupPhase(ctx, dataStore, collectionID, database, a.CompactID, a.VBUUIDs, terminator)
+			if errors.As(err, &rollbackErr) || errors.Is(err, base.ErrVbUUIDMismatch) {
```

Comment:
I'd be inclined to put the error handling into a function because we reuse for mark and cleanup, like:

handleAttachmentCompactionRollbackError(ctx, DCPClient, err) (shouldRetry, err)